From 4e9ef93c0691dab6d90935353e6de981cbddac54 Mon Sep 17 00:00:00 2001
From: Owen W. Taylor <otaylor@fishsoup.net>
Date: Wed, 27 Apr 2011 07:50:10 -0400
Subject: [PATCH] appDisplay: Fix off-by-one when incrementally adding
 application icons

A "cosmetic" code arrangement I requested in code review resulted
in one too few items being removed from the queue for each incremental
chunk of icons added. Fix.

https://bugzilla.gnome.org/show_bug.cgi?id=648739
---
 js/ui/appDisplay.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index cd502ef..6fe12a4 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -126,7 +126,7 @@ AlphabeticalView.prototype = {
             if (currentTimeMillis - startTimeMillis > MAX_APPLICATION_WORK_MILLIS)
                 break;
         }
-        this._pendingAppIds.splice(0, i);
+        this._pendingAppIds.splice(0, i + 1);
         if (this._pendingAppIds.length > 0) {
             return true;
         } else {
-- 
1.7.4.4

